include/dfu/pcd.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/** @file pcd.h
+ *
+ * @defgroup pcd Peripheral Core DFU
+ * @{
+ * @brief API for handling DFU of peripheral cores.
+ *
+ * The PCD provides functions for sending/receiving DFU images between
+ * cores on a multi core system.
+ */
+
+#ifndef PCD_H__
+#define PCD_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+#include <device.h>
+
+/** Magic value written to indicate that a copy should take place. */
+#define PCD_CMD_MAGIC_COPY 0xb5b4b3b6
+/** Magic value written to indicate that a something failed. */
+#define PCD_CMD_MAGIC_FAIL 0x25bafc15
+/** Magic value written to indicate that a copy is done. */
+#define PCD_CMD_MAGIC_DONE 0xf103ce5d
+
+/** @brief PCD command structure.
+ *
+ *  The command is used to communicate information between the sender
+ *  and the receiver of the DFU image.
+ */
+struct pcd_cmd {
```

Comment:
Could we move this inside the  C file, and only have an empty struct definition here? To protect the members, and then update the signature of the `pcd_transfer` method to instead take src, len and offset as arguments.